client.go

```diff
@@ -32,14 +33,15 @@ type Client struct {
 	Logger   logr.Logger
 	Registry *registrar.Registry
 
-	httpClient             *http.Client
-	httpClientSetupFuncs   []func(*http.Client)
-	mdLock                 *sync.Mutex
-	metadata               *bmc.Metadata
-	perProviderTimeout     func(context.Context) time.Duration
-	oneTimeRegistry        *registrar.Registry
-	oneTimeRegistryEnabled bool
-	providerConfig         providerConfig
+	httpClient                       *http.Client
+	httpClientSetupFuncs             []func(*http.Client)
+	mdLock                           *sync.Mutex
+	metadata                         *bmc.Metadata
+	perProviderTimeout               func(context.Context) time.Duration
+	oneTimeRegistry                  *registrar.Registry
+	oneTimeRegistryEnabled           bool
+	returnIncompatibleProviderErrors bool
```

Comment:
Hey @joelrebel, I like this idea! I had a similar thought a little while back. I created a branch for it but forgot to create the PR. I just opened the PR now. https://github.com/bmc-toolbox/bmclib/pull/335 It's a bit different approach. Let me know what you think. Thanks!

Reply:
Hey @jacobweinstock thanks for taking a look, your approach is better in this case and we can include that instead. I'll remove the parts that overlap in my PR.

Reply:
updated in latest commits